infra/lambda_functions/rest/deletediagram/app.js

```diff
@@ -9,11 +9,10 @@ const ddb = new AWS.DynamoDB.DocumentClient({
 });
 
 exports.handler = async (event) => {
+  const body = JSON.parse(event.body);
   const deleteParams = {
     TableName: process.env.DIAGRAMS_TABLE_NAME,
-    Key: {
-      diagramId: JSON.parse(event.body).diagramId,
-    },
+    Key: body,
```

Comment:
this sort of thing does worry me quite a bit... especially coming from a language with type safety! What is `body`? OK, we're assuming that it's a string, but what if it's json?

I would feel a lot better if we were using typescript, i.e.
```interface MyObj {
    myString: string;
    myNumber: number;
}

let obj: MyObj = JSON.parse('{ "myString": "string", "myNumber": 4 }');
console.log(obj.myString);
console.log(obj.myNumber);```